src/nnet3/nnet-diagnostics.h

```diff
@@ -108,7 +118,7 @@ class NnetComputeProb {
   void Compute(const NnetExample &eg);
 
   // Prints out the final stats, and return true if there was a nonzero count.
-  bool PrintTotalStats() const;
```

Comment:
I'm not sure what bug you refer to.  Actually I'm not sure what your whole first sentence means-- try to use less ambiguous language.
My point here is that you need to document whether tot_weight is supposed to be set, or to be added to, and what it is.
Also you have changed how the return status is computed (from or to and) and it's now inconsistent with the documentation.